### DISCUSSION TITLE
Average Time Taken?

### DISCUSSION THREAD

**Discussion by mrn1522** [winedarksea/AutoTS#221] [General]
Hello,

Just wondering for my sanity what the average time taken for AutoTS to run? Obviously it's highly dependant of many factors but using the default models list with load_monthly dataset it can take around 10 minutes. Is that in the ball park range of how long it should be taking?

**Comment by winedarksea** [comment]
10 seconds to infinite...
It really depends on

- how many series are being forecast. More series = slower
- how many generations are being run
- what models are being used
- how many cross validations are run
- whether ensembling is used
- the forecast horizon length, bigger is slower

10 minutes is on the faster end of what I would expect to see. Load_monthly is a fairly small dataset, though. And you probably only ran 5 generations or so. So 10 minutes seems reasonable.

But when forecasting 1000's or 10,000's of series, an overnight run (12+ hours) even with a faster model list is usually expected.
